extensions/HyperVExtension/src/HyperVExtension/Models/VirtualMachineCreation/VMGalleryVMCreationOperation.cs

```diff
@@ -179,34 +209,52 @@ private void UpdateProgress(string localizedString, uint percentage = 0u)
     /// <summary>
     /// Downloads the disk image from the Hyper-V VM gallery.
     /// </summary>
-    private async Task DownloadImageAsync()
+    private async Task DownloadImageAsync(VMGalleryImage image)
     {
         var downloadUri = new Uri(Image.Disk.Uri);
         var archivedFileName = _vmGalleryService.GetDownloadedArchiveFileName(Image);
         var archivedFileAbsolutePath = Path.Combine(_tempFolderSaveLocation, archivedFileName);
+        var isFileBeingDownloaded = _downloaderService.IsFileBeingDownloaded(archivedFileAbsolutePath);
 
         // If the file already exists and has the correct hash, we don't need to download it again.
-        if (File.Exists(archivedFileAbsolutePath))
+        if (File.Exists(archivedFileAbsolutePath) && !isFileBeingDownloaded)
```

Comment:
Update the comment to reflect the new if statement.

Reply:
updated thanks